handler/deployment_config_callback.go

```diff
@@ -181,14 +186,175 @@ func (h *handler) ReceiveDeleteDeploymentConfig(s *discordgo.Session, i *discord
 }
 
 func (h *handler) ReceiveDeployWithDeploymentConfig(s *discordgo.Session, i *discordgo.InteractionCreate) {
+	// TODO: replace hardcoded variables
+	host := "140.113.207.18:22"
+	user := "aisforchestrator"
+	pKey := []byte(config.Env.AisfMasterPrivateKey)
+	serverDeploymentFileRoot := "/home/aisforchestrator"
+
+	// get deployment config id
+	data := i.ModalSubmitData()
+	deploymentConfigID := data.Components[0].(*discordgo.ActionsRow).Components[0].(*discordgo.TextInput).Value
+
+	// get deployment config
+	deploymentConfig, err := h.DAO.DeploymentConfig.Get(context.Background(), deploymentConfigID)
+	if err != nil {
+		log.Println(err)
+		if err := s.InteractionRespond(i.Interaction, &discordgo.InteractionResponse{
+			Type: discordgo.InteractionResponseChannelMessageWithSource,
+			Data: &discordgo.InteractionResponseData{
+				Content: "❌ Failed to get deployment config from database",
+			},
+		}); err != nil {
+			log.Println(err)
+		}
+		return
+	}
+
+	// Create the Signer for this private key.
+	signer, err := ssh.ParsePrivateKey(pKey)
+	if err != nil {
+		fmt.Println(err.Error())
+	}
+
+	// SSH client config
+	clientConfig := &ssh.ClientConfig{
+		User:            user,
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+		Auth: []ssh.AuthMethod{
+			ssh.PublicKeys(signer),
+		},
+	}
+
+	sshClient, err := ssh.Dial("tcp", host, clientConfig)
+	if err != nil {
+		log.Println(err)
+		if err := s.InteractionRespond(i.Interaction, &discordgo.InteractionResponse{
+			Type: discordgo.InteractionResponseChannelMessageWithSource,
+			Data: &discordgo.InteractionResponseData{
+				Content: "❌ Failed to connect to server over ssh",
+			},
+		}); err != nil {
+			log.Println(err)
+		}
+		return
+	}
+	defer sshClient.Close()
+
+	// Create a new SCP client
+	scpClient, err := scp.NewClientBySSH(sshClient)
+	if err != nil {
+		log.Println(err)
+		if err := s.InteractionRespond(i.Interaction, &discordgo.InteractionResponse{
+			Type: discordgo.InteractionResponseChannelMessageWithSource,
+			Data: &discordgo.InteractionResponseData{
+				Content: "❌ Failed to connect to server over ssh",
+			},
+		}); err != nil {
+			log.Println(err)
+		}
+		return
+	}
+
+	// Connect to the remote server
+	err = scpClient.Connect()
+	if err != nil {
+		log.Println(err)
+		if err := s.InteractionRespond(i.Interaction, &discordgo.InteractionResponse{
+			Type: discordgo.InteractionResponseChannelMessageWithSource,
+			Data: &discordgo.InteractionResponseData{
+				Content: "❌ Failed to connect to server over ssh",
+			},
+		}); err != nil {
+			log.Println(err)
+		}
+		return
+	}
+	// Close client connection after the file has been copied
```

Comment:
This annotation is wrong 🥲
The client will close after the function finished.

Reply:
Moreover, try to add `defer` after creating client. (Not after error handling...
Imagine the situation that some error happened above, the client will never be closed...